mds/authent/oauthlib_utils.py

```diff
@@ -24,6 +25,16 @@ def signed_token_generator(request):
     return token
 
 
+class Backend(oauth2_provider.oauth2_backends.JSONOAuthLibCore):
+    def extract_body(self, request):
+        body = super().extract_body(request)
+        if not body:
+            body = oauth2_provider.oauth2_backends.OAuthLibCore.extract_body(
```

Comment:
Feels strange to reuse the method of another class as a function. Can't we specify a list of Backends? When one fails, it will try the next one, just like user authentication.